gbpservice/nfp/configurator/agents/loadbalancer_v1.py

```diff
@@ -0,0 +1,655 @@
+import os
+import oslo_messaging as messaging
+
+from gbpservice.nfp.configurator.agents import agent_base
+from oslo_log import log as logging
+from gbpservice.nfp.core.main import Event
+from gbpservice.nfp.configurator.lib import utils
+from gbpservice.nfp.configurator.lib.filter import Filter
+from gbpservice.nfp.core import poll as nfp_poll
+from gbpservice.nfp.configurator.lib import lb_constants
+
+LOG = logging.getLogger(__name__)
+
+
+class LBaasRpcSender(Filter):
+    """Agent side of the Agent to Plugin RPC API."""
+
+    API_VERSION = '2.0'
+    # history
+    #   1.0 Initial version
+    #   2.0 Generic API for agent based drivers
+    #       - get_logical_device() handling changed on plugin side;
+    #       - pool_deployed() and update_status() methods added;
+
+    def __init__(self, sc=None):
+        self.sc = sc
+        pass
+
+    def _notification(self, data):
+        LOG.info("Sending notification: %s" % data)
+        event = self.sc.new_event(
+            id='NOTIFICATION_EVENT', key='NOTIFICATION_EVENT', data=data)
+        self.sc.poll_event(event)
+
+    def get_logical_device(self, pool_id, context=None):
+        # Call goes to filter library
+        return self.call(
+            context,
+            self.make_msg(
+                'get_logical_device',
+                pool_id=pool_id
+            )
+        )
+
+    def update_status(self, obj_type, obj_id, status):
+        msg = {'receiver': lb_constants.NEUTRON,
+               'resource': lb_constants.SERVICE_TYPE,
+               'method': 'update_status',
+               'kwargs': [{'obj_type': obj_type,
+                           'obj_id': obj_id,
+                           'status': status}]
+               }
+        self._notification(msg)
+
+    def update_pool_stats(self, pool_id, stats):
+        LOG.info("[LbaaSRpcSender]: Update pool stats called")
```

Comment:
We are following a different way of logging for future extendibility. We frame the message and then invoke the logging API. Please check firewall/configurator for reference.